addons/test_mail/tests/test_mail_mail.py

```diff
@@ -65,6 +80,380 @@ def test_mail_mail_return_path(self):
+            self.assertFalse(notification.failure_type, 'SMTPServerDisconnected during Send raises and lead to a rollback')
+            self.assertEqual(notification.notification_status, 'ready', 'SMTPServerDisconnected during Send raises and lead to a rollback')
+
+            # MemoryError: should always raise
+            def _send_email(*args, **kwargs):
+                raise MemoryError("Some exception")
+            self.send_email_mocked.side_effect = _send_email
+
+            # MemoryError are raised even with raise_Exception=False
+            _reset_data()
+            with self.assertRaises(MemoryError), mute_logger('odoo.addons.mail.models.mail_mail'):
+                mail.send(raise_exception=False)
+            self.assertFalse(mail.failure_reason, 'MemoryError during Send raises and lead to a rollback')
+            self.assertFalse(mail.failure_type, 'MemoryError during Send raises and lead to a rollback')
+            self.assertEqual(mail.state, 'outgoing', 'MemoryError during Send raises and lead to a rollback')
+            self.assertFalse(notification.failure_type, 'MemoryError during Send raises and lead to a rollback')
+            self.assertEqual(notification.notification_status, 'ready', 'MemoryError during Send raises and lead to a rollback')
+
+            # MailDeliveryException: should be catched
+            def _send_email(*args, **kwargs):
+                raise MailDeliveryException("Some exception")
+            self.send_email_mocked.side_effect = _send_email
+
+            _reset_data()
+            with mute_logger('odoo.addons.mail.models.mail_mail'):
+                mail.send(raise_exception=False)
+            self.assertEqual(mail.failure_reason, 'Some exception')
+            self.assertFalse(mail.failure_type, 'Mail: unlogged failure type to fix')
+            self.assertEqual(mail.state, 'exception')
+            self.assertEqual(notification.failure_type, 'unknown', 'Mail: generic failure type')
+            self.assertEqual(notification.notification_status, 'exception')
+
+            # Other issues are sub-catched under a MailDeliveryException and are catched
+            def _send_email(*args, **kwargs):
+                raise ValueError("Unexpected issue")
+            self.send_email_mocked.side_effect = _send_email
+
+            _reset_data()
+            with mute_logger('odoo.addons.mail.models.mail_mail'):
+                mail.send(raise_exception=False)
+            self.assertEqual(mail.failure_reason, 'Unexpected issue')
+            self.assertFalse(mail.failure_type, 'Mail: unlogged failure type to fix')
+            self.assertEqual(mail.state, 'exception')
+            self.assertEqual(notification.failure_type, 'unknown', 'Mail: generic failure type')
+            self.assertEqual(notification.notification_status, 'exception')
+
+            self.send_email_mocked.side_effect = _send_current
```

Comment:
those tests look very redundant, can't make a list of Exception / state / failure_type ?